### PR TITLE
pcre: enable utf-8

### PR DESCRIPTION
**Pull request by musfay**
###### Motivation for this change
`pcre` is required for regular expressions in the Search & Replace feature of the bulk renamer in thunar but this feature is disabled with probably lots of languages because of `pcre` is not compiled with UTF-8 support. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
This looks good to me, I'm leaving this open until monday to give more people a chance to have a look.

File: pkgs/development/libraries/pcre/default.nix [hunk 1]
```diff
@@ -27,6 +27,7 @@ in stdenv.mkDerivation {
   configureFlags = optional (!stdenv.hostPlatform.isRiscV && !(stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64)) "--enable-jit" ++ [
     "--enable-unicode-properties"
     "--disable-cpp"
+    "--enable-utf8"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/pcre/default.nix:30]: Does this just work on all arches or do we need some special handling?

**Comment by happysalada** [comment]
I was just looking at the docs
https://www.pcre.org/original/doc/html/pcrebuild.html#SEC6
it doesn't mention anything particular about architectures.
However the flag seems to be `enable-utf` no 8 in the end. Are you sure that the flag is correct?

File: pkgs/development/libraries/pcre/default.nix [hunk 2]
```diff
@@ -27,6 +27,7 @@ in stdenv.mkDerivation {
   configureFlags = optional (!stdenv.hostPlatform.isRiscV && !(stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64)) "--enable-jit" ++ [
     "--enable-unicode-properties"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/pcre/default.nix:28]: According to the docs here https://www.pcre.org/original/doc/html/pcrebuild.html#SEC7 this option already enforces ``--enable-utf8``. I think we can close this.

**Comment by musfay** [comment]
Thunar probably have another problem if `pcre` is ok. Closing this.

**Comment by musfay** [comment]
After a regular update and garbage collection it started to work. I don't know actually what happened but it works :)

**Pull request closed without merging**